auth/token.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"net/http"
+	"time"
+)
+
+type Token struct {
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	Scope        string `json:"scope"`
+	IDToken      string `json:"id_token"`
+	TokenType    string `json:"token_type"`
+	ExpiresIn    int    `json:"expires_in"`
+	Expiry       time.Time
+}
+
+func (t *Token) SetAuthHeader(r *http.Request) {
+	r.Header.Set("Authorization", "Bearer "+t.AccessToken)
```

Comment:
```suggestion
	r.Header.Set("Authorization", fmt.Sprintf("Bearer %s", t.AccessToken))
```

Reply:
string concatenation via `+` is way faster (https://dev.to/pmalhaire/concatenate-strings-in-golang-a-quick-benchmark-4ahh) given how simple the string is (appended at the end) I don't think formatting here is worth it

Reply:
Can we really use this article as proof that this is correct? looking at the comments many developers don’t agree. For instance:

![Screenshot 2022-01-25 at 12 26 52](https://user-images.githubusercontent.com/5663078/150976976-f3d3adc1-07a8-49a6-9716-dbaeca9e05b5.png)

Anyway, this is not blocking at all : ). Thanks for sharing the article

Reply:
yes, if you read that comment plus is still faster than `Sprintf`